react/Breadcrumb.tsx

```diff
@@ -1,43 +1,41 @@
-import PropTypes from 'prop-types'
-import React, { Component, Fragment, ReactNode } from 'react'
+import React, { Fragment, useMemo } from 'react'
 import { Link } from 'vtex.render-runtime'
 import unorm from 'unorm'
 import { IconHome, IconCaret } from 'vtex.store-icons'
 
-import breadcrumb from './breadcrumb.css'
+import styles from './breadcrumb.css'
 
-const LINK_CLASS_NAME = `${breadcrumb.link} dib pv1 link ph2 c-muted-2 hover-c-link`
+const LINK_CLASS_NAME = `${styles.link} dib pv1 link ph2 c-muted-2 hover-c-link`
 
-interface DefaultProps {
-  categories: Array<string>
+interface Category {
+  name: string
+  link: string
 }
-  
-interface Props extends DefaultProps {
+interface Props {
   term?: string
-}
-
-type Category = {
-  name: string
-  value: string
+  categories: Array<string>
 }
 
 /**
- * Breadcrumb Component.
+ * Return a new string with the first letter as capital
+ * @param string:  String to be capitalized
  */
-class Breadcrumb extends Component<Props> {
-  public static readonly defaultProps = {
-    categories: [],
-  }
+const capitalize = (string: string): string =>
+  string.charAt(0).toUpperCase() + string.slice(1).toLocaleLowerCase()
 
-  public static readonly propTypes = {
-    /** Search term or product slug. */
-    term: PropTypes.string,
-    /** Product's categories. */
-    categories: PropTypes.arrayOf(PropTypes.string),
-  }
+/**
+ * Returns a category name by replacing the hypens with spaces and capitalizing
+ * the result
+ * @param categoryKey
+ */
+const getCategoryName = (categoryKey: string): string =>
+  capitalize(categoryKey.replace(/[-]/g, ' '))
```

Comment:
The real issue is not about capitalizing the words.

See the props that breadcrumbs uses:

![image](https://user-images.githubusercontent.com/284515/54963313-17147600-4f47-11e9-9f06-a465a8aac303.png)

The issue is that we are getting those urls and transforming them to the breadcrumb values transforming this string.

The fix should be: make Breadcrumb get the category names as it is written in the Catalog system and print them as they are typed in the system. We should not run any string transformation on our side.

Reply:
Yep, you are definitely right. What you said was my original approach to this problem, but I found some issues doing that (you can see it in a [PR that I closed ](https://github.com/vtex-apps/breadcrumb/pull/34)).

The approach was to receive something like:
```
{ 
  categories : [
    { name: 'Camisetas Masculinas', link: '/Camisetas-Masculinas/' },
    { name: 'Algodão', link: '/Camisetas-Masculinas/Algodao'  } 
  ],
  term : 'Camiseta de Algodão Branca' 
}
```

In the case of `search-result`, it works fine because I can access the `searchQuery` and obtain the names. The problem is towards the `product-details` page. The `store` contains the context that wraps the page, which is`ProductContext`. 
The [product query](https://github.com/vtex-apps/store-resources/blob/master/react/queries/product.gql) (used in the context) get some categories with the shape: `["/Camisetas-Masculinas/Algodao" 
"/Camisetas-Masculinas/"]`, and do not even retrieve the names so I can assembly the categories the right way.
The best solution for this problem would already receive the categories in the right shape or at least the categories names.

This PR just do the right transformations to the name, acting as a workaround. This way, the "lowercased-hyphen" categories look better and gives more time to better solve this problem.